sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/StreamingJoinHelper.scala

```diff
@@ -232,6 +232,7 @@ object StreamingJoinHelper extends PredicateHelper with Logging {
           collect(left, negate) ++ collect(right, !negate)
         case TimeAdd(left, right, _) =>
           collect(left, negate) ++ collect(right, negate)
+        case DatetimeSub(_, _, child) => collect(child, negate)
```

Comment:
Just to check; did the change in Analyzer (https://github.com/apache/spark/pull/28402/files#diff-57b3d87be744b7d79a9beacf8e5e5eb2R310-R313) invoke some test failures in streaming joins?

Reply:
https://github.com/apache/spark/commit/beec8d535f093f1867678fe5afeb02453464f90d#diff-7c2a9f40678db8001b80d39a095dab31L235-L236

DatetimeSub is justing taking the place of `TimeAdd` here